mne/rank.py

```diff
@@ -413,7 +413,10 @@ def _compute_rank(
         info = info.copy()
         info["bads"] = []
         inst = pick_channels_cov(
-            inst, set(inst["names"]) & set(info["ch_names"]), exclude=[], ordered=False
+            inst,
+            set(inst["names"]) & set(info["ch_names"]),
+            exclude=info["bads"] + inst["bads"],
+            ordered=False,
```

Comment:
cc @mscheltienne by setting `info["bads"] = []` here bad channels were errantly be included in the cov during calculation as of #9719 in some circumstances